portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,106 @@
+package com.google.sps.servlets;
+
+import com.google.sps.data.ContactMe;
+import com.google.gson.Gson;
+ 
+import java.io.FileWriter;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+ 
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+
+@WebServlet("/form-handler")
+public class FormHandlerServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Get the value entered in the form.
+    String name = request.getParameter("name");
+    String email = request.getParameter("email");
+    String reason = request.getParameter("reason");
+    String recruiter = request.getParameter("recruiter");
+
+    //Check for nullity in rec
+    if(recruiter == null) {
+      recruiter = "null";
+    }
+
+    // Convert the server stats to JSON
+    ContactMe contactMe = new ContactMe(name, email, reason, recruiter);
+    String jsonContactMe = convertToJson(contactMe);
+    System.out.println(jsonContactMe);
+
+    //Read previous forms
+    JSONArray contactList = readFile();
+    System.out.println(contactList);
+    //Add new contact to list
+    contactList.add(jsonContactMe);
+
+    //Write JSON file
+    try (FileWriter file = new FileWriter("src/main/java/com/google/sps/db/contactMe.json")) {
```

Comment:
Thanks I will log it out! I see it is a multi thread server, this may cause the issue that if 2+ people summit the file at the same time some information could be lost (only 1 contact saved for example). I think the best solution will be next week that we will add databases.